lib/IRGen/GenCall.cpp

```diff
@@ -866,10 +866,11 @@ llvm::Type *SignatureExpansion::expandExternalSignatureTypes() {
   // Swift parameters list?
   size_t clangToSwiftParamOffset = paramTys.size();
 
-  // Convert each parameter to a Clang type.
+  // Convert each parameter to a canonical Clang parameter type.
   for (auto param : params) {
     auto clangTy = IGM.getClangType(param);
-    paramTys.push_back(clangTy);
+    auto canClangTy = IGM.getClangASTContext().getCanonicalParamType(clangTy);
+    paramTys.push_back(canClangTy);
```

Comment:
I think the better fix here might be to call getUnqualifiedType in ClangTypeConverter::convert's code for getting the type out of a clang::TypeDecl.

Reply:
Ok, I'll take a look tomorrow.

Reply:
Ok, that made the change a bit simpler. PTAL